lib/calc_route.dart

```diff
@@ -0,0 +1,137 @@
+
+    return new Cart_Point(x, y);
+  }
+}
+
+bool check_viable_tack(Cart_Point p1, Cart_Point p2, double wind_radians) {
+  /*
+    check_viable_tack just checks if a tack from p1 -> p2 is makable (not directly upwind).
+    Args:
+        p1 (Cart_Point): Cartesian coordinates of start
+        p2 (Cart_Point): Cartesian coordinates of end
+        wind_radians (double): wind direction 
+    Returns:
+        bool
+  */
+  double no_go_limit_radians = 0.610865; // Equivilant to 35 degrees
+
+  double ngzone_upper_bound = (wind_radians + no_go_limit_radians);
+  double ngzone_lower_bound = (wind_radians - no_go_limit_radians);
+  verbose_print("wind lower-upper: " + radToDegStr(ngzone_lower_bound) + " <-> " + radToDegStr(ngzone_upper_bound));
+  Point delta = p2.toPoint() - p1.toPoint();
+  double tack_angle = atan2(delta.y, delta.x);
+  verbose_print("pot viable tack: " + radToDegStr(tack_angle));
+  // arctan gives values fro [-pi, pi]
+  if (tack_angle < 0) {
+    tack_angle += 2 * pi;
+  }
+
+  verbose_print("(tack_angle > ngzone_lower_bound) => " + (tack_angle > ngzone_lower_bound).toString());
+  verbose_print("(tack_angle < ngzone_upper_bound) => " + (tack_angle < ngzone_upper_bound).toString());
+  return !((tack_angle > ngzone_lower_bound) && (tack_angle < ngzone_upper_bound));
+}
+
+String radToDegStr(double radians) => (radians * 57.2958).toString();
+
+double bestTackAngle(Cart_Point start, Cart_Point goal, double wind_radians, bool invert_flag) {
+  double no_go_limit_radians = 0.610865; // Equivilant to 35 degrees
+  Point delta = goal.toPoint() - start.toPoint();
+  double impossible_tack = atan2(delta.y, delta.x);
+  verbose_print("imposs_tack  =  " + radToDegStr(impossible_tack));
+  double ngzone_upper_bound = (wind_radians + no_go_limit_radians);
+  double ngzone_lower_bound = (wind_radians - no_go_limit_radians);
+  verbose_print("wind lower-upper: " + radToDegStr(ngzone_lower_bound) + " <-> " + radToDegStr(ngzone_upper_bound));
+
+  var angle_from_ideal_upper = (impossible_tack - ngzone_upper_bound);
+  var angle_from_ideal_lower = (impossible_tack - ngzone_lower_bound);
+
+  verbose_print("possible tacks: " + radToDegStr(angle_from_ideal_lower) + " <-> " + radToDegStr(angle_from_ideal_upper));
+  bool lower_better = (angle_from_ideal_lower < angle_from_ideal_upper);
+  lower_better = (invert_flag) ? !lower_better : lower_better;
+
+  return (lower_better) ? ngzone_lower_bound : ngzone_upper_bound;
+}
+
+calculateOptimal(RouteModel route) {
+  List intermediate_points = new List<Cart_Point>();
+  bool viable = check_viable_tack(route.start, route.end, route.wind_radians);
+  if (!viable) {
+    double best_tack_slope = bestTackAngle(route.start, route.end, route.wind_radians, false);
+
+    verbose_print("__BEST TACK " + radToDegStr(best_tack_slope));
+    // TODO this can be done better, note th +pi
+    double inverse_tack_slope = bestTackAngle(route.end, route.start, route.wind_radians, true) + pi;
```

Comment:
https://www.dartlang.org/guides/language/effective-dart/style I think lowerCamelCase for inverse_tack_slope